src/cartridges/int_adyen_SFRA/cartridge/client/default/js/amazon.js

```diff
@@ -0,0 +1,79 @@
+const store = require('../../../store');
+
+const amazonPayNode = document.getElementById('amazon-container');
+
+function handleAmazonResponse(response, component) {
+  if (response.fullResponse?.action) {
+    component.handleAction(response.fullResponse.action);
+  } else if (response.resultCode === 'Authorised') {
+    document.querySelector('#result').value = JSON.stringify({
+      pspReference: response.fullResponse.pspReference,
+      resultCode: response.fullResponse.resultCode,
+      paymentMethod: response.fullResponse.additionalData.paymentMethod,
+    });
+    document.querySelector('#showConfirmationForm').submit();
+  } else if (response.error) {
+    document.querySelector('#result').value = JSON.stringify({
+      error: true,
+    });
+    document.querySelector('#showConfirmationForm').submit();
+  }
+}
+
+function paymentFromComponent(data, component) {
+  $.ajax({
+    url: window.paymentFromComponentURL,
+    type: 'post',
+    data: {
+      data: JSON.stringify(data),
+      paymentMethod: 'amazonpay',
+    },
+    success(response) {
+      if (response.orderNo) {
+        document.querySelector('#merchantReference').value = response.orderNo;
+      }
+      handleAmazonResponse(response, component);
+    },
+  }).fail(() => {});
```

Comment:
This fail response is empty.
I think either the response needs to be handled in terms of an error message or the fail clause should be removed